packages/css/src/rules/utils.ts

```
@@ -10,3 +18,166 @@ export function mapValues<Input extends Record<string, unknown>, OutputValue>(
+>(
+  variants?:
+    | VariantSelection<Variants & ToggleVariantMap<ToggleVariants>>
+    | undefined
+): VariantObjectSelection<Variants & ToggleVariantMap<ToggleVariants>> {
+  if (Array.isArray(variants)) {
+    return variants.reduce<
+      VariantObjectSelection<Variants & ToggleVariantMap<ToggleVariants>>
+    >((acc, variant) => {
+      if (typeof variant === "string") {
+        return {
+          ...acc,
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)